test/ably/resthttp_test.py

```diff
@@ -135,3 +135,28 @@ def test_custom_http_timeouts(self):
         self.assertEqual(ably.http.http_open_timeout, 8)
         self.assertEqual(ably.http.http_max_retry_count, 6)
         self.assertEqual(ably.http.http_max_retry_duration, 20)
+
+    def test_no_host_fallback_if_environment_is_set(self):
+        # RSC15b
+        ably = AblyRest(token="foo", environment="sandbox")
+        self.assertEqual(ably.options.fallback_hosts, [])
```

Comment:
I am not really sure I agree that this is testing the fallback functionality.  The thinking is simply that nowhere in the spec does it state that `AblyRest.options.fallback_hosts` is an attribute, and nowhere is that attribute linked to the fallback behaviour, you have made that assumption based on the inner workings of the library.  So if someone refactors this code such that the `options.fallback_hosts` attribute is no longer used to drive the fallback behaviour, then this test will likely continue to parse.  The spec `RSC15b` that you are referencing states `If host fallback is not supported, failing HTTP requests that would have qualified for a retry against a fallback host, will instead result in an error immediately`, yet you have not tested this anywhere.  You should at a minimum write a test checks that when fallback hosts are set in `ably.options.fallback_hosts`, then those hosts are tried when the primary endpoints fails, and then have a test that does a test where `ably.options.fallback_hosts` is an empty array, and in that instance, tries a request that fails and does not try any fallback hosts.  Once you have those two tests in place, I would agree the permutations below may be sufficient.